skimage/morphology/misc.py

```diff
@@ -235,3 +243,208 @@ def remove_small_holes(ar, area_threshold=64, connectivity=1, *, out=None):
+    label_image,
+    min_distance,
+    *,
+    priority=None,
+    p_norm=2,
+    spacing=None,
+    out=None,
+):
+    """Remove objects until a minimal distance is ensured.
+
+    Iterates over all objects (pixels that aren't zero) inside an image and
+    removes "nearby" objects until all remaining ones are spaced more than a
+    given minimal distance from each other.
+
+    Parameters
+    ----------
+    label_image : ndarray of integers
+        An n-dimensional array containing object labels, e.g. as returned by
+        :func:`~.label`. A value of zero is considered background, all other
+        object IDs must be positive integers.
+    min_distance : int or float
+        Remove objects with lower priority whose distance is not greater than
+        this positive value.
+    priority : ndarray, optional
+        Defines the priority with which objects that are too close to each other
+        are removed. Expects a 1-dimensional array of length
+        :func:`np.amax(label_image) + 1 <numpy.amax>` that contains the priority
+        for each object ID at the respective index. Objects with a lower value
+        are removed first until all remaining objects fulfill the distance
+        requirement. If not given, priority is given to objects with a higher
+        number of samples and their ID second.
+    p_norm : int or float, optional
+        The Minkowski p-norm, used to calculate the distance between objects.
+        The default ``2`` corresponds to the Euclidean distance, ``1`` to the
+        "Manhattan" distance, and ``np.inf`` to the Chebyshev distance.
+    spacing : sequence of float, optional
+        The pixel spacing along each axis of `label_image`. If not specified,
+        a grid spacing of unity (1) is implied.
+    out : ndarray, optional
+        Array of the same shape and dtype as `image`, into which the output is
+        placed. By default, a new array is created.
+
+    Returns
+    -------
+    out : ndarray
+        Array of the same shape as `label_image`, for which objects that violate
+        the `min_distance` condition were removed.
+
+    See Also
+    --------
+    skimage.morphology.remove_small_objects
+        Remove objects smaller than the specified size.
+
+    Notes
+    -----
+    The basic steps of this algorithm work as follows:
+
+    1. If `priority` is not given, use :func:`numpy.bincount` as a fallback.
+    2. Find the indices for of all given objects and separate them depending on
+       if they point to an object's border or not.
+    3. Sort indices by their object ID, ensuring all indices pointing to the
+       same object are next to each other. This optimization allows finding
+       all parts of an object, simply by stepping to the neighboring indices.
+    4. Sort boundary indices by priority. Use a stable-sort to preserve the
+       contiguousness from the previous sorting step.
+    5. Construct a kd-tree with :func:`scipy.spatial.cKDTree` from the
```

Comment:
```suggestion
    5. Construct a :func:`scipy.spatial.cKDTree <kd-tree>` from the
```

Not sure if this is the right syntax, but you get the idea.